src/proxy-animation.js

```diff
@@ -951,32 +950,32 @@ function parseAnimationRange(timeline, value) {
     //                                        <name> <end-offset>
     // <start-offset> <end-offset> --> cover <start-offset> cover <end-offset>
     // TODO: Support all formatting options once ratified in the spec.
-    const parts = value.split(/\s+/);
+    const parts = splitIntoComponentValues(value);
     const rangeNames = [];
     const offsets = [];
-  
+
     parts.forEach(part => {
-      if (part.endsWith('%'))
-        offsets.push(parseFloat(part));
-      else
+      if (ANIMATION_RANGE_NAMES.includes(part)) {
         rangeNames.push(part);
+      } else {
+        offsets.push(CSSNumericValue.parse(part));
+      }
```

Comment:
This piece of code can throw an exception.

For example, when an author sets the range to `'invalid 100%'`, the first component will be `'invalid'` and the second `'100%'`. When processing `'invalid'` the logic will fall through to the `CSSNumericValue.parse(part)` part, upon which it will throw an exception.

Further down below in the code there’s already little piece of similar code that catches the exception and throws a `TypeError`. It might be a good idea to do this here as well, so that it makes things easier for authors to debug.

```suggestion
      if (ANIMATION_RANGE_NAMES.includes(part)) {
        rangeNames.push(part);
      } else {
        try {
          offsets.push(CSSNumericValue.parse(part));
        } catch (e) {
          throw TypeError(`Could not parse range "${value}"`);
        }
      }
```